clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -2816,8 +2817,16 @@ private void handleUpdateRequest(
         readerValueSchemaId,
         consumerRecord.getTopicPartition());
 
-    // Apply Write Compute.
     final byte[] updatedValueBytes;
+    // Samza VeniceWriter doesn't handle chunking config properly. It reads chunking config
+    // from user's input instead of getting it from store's metadata repo. This causes SN
+    // to der-se of keys a couple of times.
+    final byte[] updatedKeyBytes =
+        isChunked ? ChunkingUtils.KEY_WITH_CHUNKING_SUFFIX_SERIALIZER.serializeNonChunkedKey(keyBytes) : keyBytes;
+    final ChunkedValueManifest oldValueManifest = isChunked
```

Comment:
Same comment as `ActiveActiveStoreIngestionTask` since the previous manifest should be already available.